packages/react-native-usb/android/src/main/java/io/trezor/rnusb/ReactNativeUsbModule.kt

```diff
@@ -234,15 +234,17 @@ class ReactNativeUsbModule : Module() {
 
     private fun closeDevice(deviceName: String) {
         Log.d(LOG_TAG, "Closing device $deviceName")
-        val usbConnection = getOpenedConnection(deviceName)
-        usbConnection.close()
+        getOpenedConnection(deviceName).close()
         openedConnections.remove(deviceName)
     }
 
     private fun closeAllOpenedDevices() {
         Log.d(LOG_TAG, "Closing all devices")
-        openedConnections.forEach { (deviceName, _) ->
-            closeDevice(deviceName)
+        with(openedConnections.iterator()) {
+            forEach {
+                it.value.close()
```

Comment:
Can you please explain little bit more why this is better than previous solution? Both seems pretty similar and I don't get it why one causes concurency issues and one not.

Reply:
In Java/Kotlin, it’s illegal to remove an entry from a map while iterating over it using a for loop. One needs to use iterator or some alternative approaches, see https://www.baeldung.com/kotlin/map-remove-while-iterating for further details.

Reply:
The first commit contains alternative syntax, if you prefer it more.

Reply:
Thanks for explanation now it makes more sense to me. Second version looks better IMHO.